extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -419,21 +419,33 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 
 	async findAndExpandParentNode(notebookPath: string): Promise<BookTreeItem> {
 		let bookItem: BookTreeItem = this.currentBook?.getNotebook(notebookPath);
-		// if the node is not expanded getNotebook returns undefined, try to expand the parent node or getChildren of
-		// the root node.
-		if (!bookItem) {
-			// get the parent node and expand it if it's not already
-			let allNodes = this.currentBook?.getAllNotebooks();
-			let book = allNodes ? Array.from(allNodes?.keys())?.filter(x => x.indexOf(notebookPath.substring(0, notebookPath.lastIndexOf(path.sep))) > -1) : undefined;
-			let bookNode = book?.length > 0 ? this.currentBook?.getNotebook(book.find(x => x.substring(0, x.lastIndexOf(path.sep)) === notebookPath.substring(0, notebookPath.lastIndexOf(path.sep)))) : undefined;
-			if (bookNode) {
-				if (this._bookViewer?.visible) {
-					await this._bookViewer.reveal(bookNode, { select: true, focus: false, expand: 3 });
-				} else {
-					await this.getChildren(bookNode);
+		this.currentBook = this.books.length > 1 ? this.books.find(b => notebookPath.indexOf(b.bookPath) > -1) : this.currentBook;
+		// Try to expand the parent node until we get to the notebook
+		if (!bookItem && this.currentBook) {
+			// get the children of root node and expand the nodes to the notebook level.
+			await this.getChildren(this.currentBook.rootNode);
+			// number of levels to expand
+			let depthOfNotebookInBook: number = path.relative(notebookPath, this.currentBook.bookPath)?.split(path.sep)?.length ?? 0;
+			if (this.currentBook.bookItems.length > 0 && depthOfNotebookInBook > 0) {
+				while (depthOfNotebookInBook > 0) {
+					// check if the notebook is available in already expanded levels.
+					bookItem = this.currentBook.bookItems.find(b => b.tooltip === notebookPath);
+					if (bookItem) {
+						break;
```

Comment:
Don't we still want to reveal in this case? This would be for the case where a user expanded to the node and then collapsed its parent so it's not actually visible anymore (but we still loaded it)